### ISSUE TITLE
[PATCH] Allow CORE::* without feature.pm

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#11321]
Migrated from [rt.perl.org#90130](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=90130) (status was 'resolved')

Searchable as RT90130$

**Comment by p5pRT** [comment]
### From @cpansprout

Is there any reason feature\.pm\-enabled keywords should not be available via CORE&#8203;::foo syntax even without feature\.pm loaded?

The attached 1\-character patch enables those\.

If this is acceptable\, I’ll write tests\, too\.



---
Flags&#8203;:
&nbsp;    category=core
&nbsp;    severity=low


---
Site configuration information for perl 5\.14\.0&#8203;:

Configured by sprout at Thu May  5 13&#8203;:28&#8203;:18 PDT 2011\.

Summary of my perl5 \(revision 5 version 14 subversion 0\) configuration&#8203;:
&nbsp;  Snapshot of&#8203;: f5a93a43fe9802cca88dfd56d68a220ec6625118
&nbsp;  Platform&#8203;:
&nbsp;    osname=darwin\, osvers=10\.5\.0\, archname=darwin\-thread\-multi\-2level
&nbsp;    uname='darwin pint\.local 10\.5\.0 darwin kernel version 10\.5\.0&#8203;: fri nov 5 23&#8203;:20&#8203;:39 pdt 2010; root&#8203;:xnu\-1504\.9\.17~1release\_i386 i386 '
&nbsp;    config\_args='\-Dusedevel \-de \-Duseithreads \-Doptimize=\-g'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=define\, usemultiplicity=define
&nbsp;    useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
&nbsp;    use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, ccflags ='\-fno\-common \-DPERL\_DARWIN \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include'\,
&nbsp;    optimize='\-g'\,
&nbsp;    cppflags='\-fno\-common \-DPERL\_DARWIN \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='4\.2\.1 \(Apple Inc\. build 5664\)'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='env MACOSX\_DEPLOYMENT\_TARGET=10\.3 cc'\, ldflags =' \-fstack\-protector \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib
&nbsp;    libs=\-ldbm \-ldl \-lm \-lutil \-lc
&nbsp;    perllibs=\-ldl \-lm \-lutil \-lc
&nbsp;    libc=/usr/lib/libc\.dylib\, so=dylib\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=bundle\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags=' '\, lddlflags=' \-bundle \-undefined dynamic\_lookup \-L/usr/local/lib \-fstack\-protector'

Locally applied patches&#8203;:
&nbsp;    RC2



---
@&#8203;INC for perl 5\.14\.0&#8203;:
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.14\.0/darwin\-thread\-multi\-2level
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.14\.0
&nbsp;    /usr/local/lib/perl5/5\.14\.0/darwin\-thread\-multi\-2level
&nbsp;    /usr/local/lib/perl5/5\.14\.0
&nbsp;    /usr/local/lib/perl5/site\_perl
&nbsp;    \.



---
Environment for perl 5\.14\.0&#8203;:
&nbsp;    DYLD\_LIBRARY\_PATH \(unset\)
&nbsp;    HOME=/Users/sprout
&nbsp;    LANG=en\_US\.UTF\-8
&nbsp;    LANGUAGE \(unset\)
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)
&nbsp;    PATH=/usr/bin&#8203;:/bin&#8203;:/usr/sbin&#8203;:/sbin&#8203;:/usr/local/bin&#8203;:/usr/X11/bin&#8203;:/usr/local/bin
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From @cpansprout

<details><summary>Inline Patch</summary>

```diff
diff --git a/toke.c b/toke.c
index 0f08d42..1b581f7 100644
--- a/toke.c
+++ b/toke.c
@@ -7055,7 +7055,7 @@ Perl_yylex(pTHX)
 		s += 2;
 		d = s;
 		s = scan_word(s, PL_tokenbuf, sizeof PL_tokenbuf, FALSE, &len);
-		if (!(tmp = keyword(PL_tokenbuf, len, 0)))
+		if (!(tmp = keyword(PL_tokenbuf, len, 1)))
 		    Perl_croak(aTHX_ "CORE::%s is not a keyword", PL_tokenbuf);
 		if (tmp < 0)
 		    tmp = -tmp;
```
</details>

**Comment by p5pRT** [comment]
### From @ikegami

On Sun\, May 8\, 2011 at 5&#8203;:53 PM\, Father Chrysostomos \<
perlbug\-followup@&#8203;perl\.org> wrote&#8203;:

> The attached 1\-character patch enables those\.
>

Don't be so modest\, it's a 1\-bit patch\! :\)

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @cpansprout

On Sun May 08 14&#8203;:53&#8203;:21 2011\, sprout wrote&#8203;:
> Is there any reason feature\.pm\-enabled keywords should not be
> available via CORE&#8203;::foo syntax even without feature\.pm loaded?
> 
> The attached 1\-character patch enables those\.
> 
> If this is acceptable\, I’ll write tests\, too\.

I’ve now applied an expanded version of that patch as 9dcb836\.

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

On Sun May 08 14&#8203;:53&#8203;:21 2011\, sprout wrote&#8203;:
> Is there any reason feature\.pm\-enabled keywords should not be
> available via CORE&#8203;::foo syntax even without feature\.pm loaded?
> 
> The attached 1\-character patch enables those\.
> 
> If this is acceptable\, I’ll write tests\, too\.

I’ve now applied an expanded version of that patch as 9dcb836\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
